src/bin/cli/solana/deploy.rs

```diff
@@ -0,0 +1,113 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::cli::check_target_match;
+use std::{process::exit, str::FromStr, time::Duration};
+
+use solana_cli::{
+    cli::{
+        process_command, CliCommand, CliCommandInfo, CliConfig, DEFAULT_CONFIRM_TX_TIMEOUT_SECONDS,
+        DEFAULT_RPC_TIMEOUT_SECONDS,
+    },
+    program::ProgramCliCommand,
+};
+use solana_cli_config::{Config, CONFIG_FILE};
+use solana_cli_output::OutputFormat;
+use solana_rpc_client_api::config::RpcSendTransactionConfig;
+use solana_sdk::{commitment_config::CommitmentConfig, signer::keypair::read_keypair_file};
+
+#[derive(Clone, Debug, clap::Args)]
+#[clap(name = "deploy", about = "Deploy a program to the Solana cluster")]
+pub struct SolanaDeploy {
+    #[clap(help = "Specifies the path to the program file to deploy (.so)")]
+    program_location: String,
+    #[clap(long, help = "Specifies whether to export the output in JSON format")]
+    output_json: bool,
+    #[clap(
+        short('v'),
+        long,
+        conflicts_with = "output_json",
+        help = "Specifies whether to display verbose program deployment information"
+    )]
+    verbose: bool,
```

Comment:
The option here seem a bit sparse. For example, shouldn't there be a way to specify network endpoints?

Reply:
We are currently relying on the default `solana` local configuration file. Solana offers CLI commands for managing this configuration file, as detailed in the [CLI commands documentation](https://docs.solana.com/cli/usage#solana-config). 
However, we can also consider adding parameters into `solang` CLI to override these configurations. 
Would you recommend this approach?

Reply:
Ok to me if it's configurable somehow. Can you please mention this in the docs?